src/context/AuthContext.tsx

```diff
@@ -0,0 +1,87 @@
+'use client';
+
+import React, {
+  createContext,
+  useState,
+  useContext,
+  useEffect,
+  ReactNode,
+} from 'react';
+
+import { getUser } from '../client';
+import { User } from '../types';
+
+type AuthContextProps = {
+  user: User | null;
+  login: (token: string) => void;
+  logout: () => void;
+};
+
+export const AuthContext = createContext<AuthContextProps | undefined>(
+  undefined,
```

Comment:
I would go for null. It might be better for debugging when you want differentiate between non-existing context (undefined) and the default value "null"

Actually, in javascript project, I would strongly suggest going for "null" as your default value everywhere. Cause undefined and null have a bit different semantic meaning. Undefined is "no value", and null is "empty value". 

But since in typescript using null would mean add a lot of `| null` everywhere and optional parameters would no longer would it make no sense, it would be too annoying for a little added benefit.

Reply:
I see. And null is an object in javaScript, but it is not actually! :D